src/main/java/core/basesyntax/storage/Storage.java

```
@@ -0,0 +1,29 @@
+package core.basesyntax.storage;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private final Map<String, Integer> storage = new HashMap<>();
+
+    public void put(String fruit, int quantity) {
+        storage.put(fruit, quantity);
```

put(fruit, quantity) does not validate inputs. You must validate that `fruit` is non-null/non-blank and `quantity >= 0` before calling storage.put. The checklist expects services and DAOs to validate arguments and throw informative RuntimeException (e.g., FruitShopException) on invalid input to fail fast.